packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -0,0 +1,349 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { assert } from "@fluidframework/core-utils";
+import type {
+	FieldMapObject,
+	JsonableTree,
+	SharedTreeContentSnapshot,
+	TreeFieldStoredSchema,
+	TreeNodeStoredSchema,
+} from "@fluidframework/tree/internal";
+import {
+	EmptyKey,
+	LeafNodeStoredSchema,
+	MapNodeStoredSchema,
+	ObjectNodeStoredSchema,
+} from "@fluidframework/tree/internal";
+import type { SharedTreeLeafNode, VisualSharedTreeNode } from "./VisualSharedTreeTypes.js";
+import { type VisualChildNode, VisualNodeKind, type VisualValueNode } from "./VisualTree.js";
+
+/**
+ * Converts the output of {@link sharedTreeVisualizer} to {@link VisualChildNode} type containing `schema` and `children` fields.
+ */
+export function mapToVisualChildNode(tree: VisualSharedTreeNode): VisualChildNode {
+	if ("value" in tree) {
```

Comment:
Let's clean this up. I would recommend adding a "type" or "kind" property to your tree format, then we can just switch on that here.